util.py

```diff
@@ -469,20 +470,27 @@ def get_instance_id(agentConfig):
 
 
 class Watchdog(object):
-    """Simple signal-based watchdog that will scuttle the current process
-    if it has not been reset every N seconds, or if the processes exceeds
-    a specified memory threshold.
+    """
+    Simple signal-based watchdog. Restart the process when:
+    * no reset was made for more than a specified duration
+    * (optional) a specified memory threshold is exceeded
+    * (optional) a suspicious high activity is detected, i.e. too many resets for a given timeframe.
+
+    **Warning**: Not thread-safe.
     Can only be invoked once per process, so don't use with multiple threads.
     If you instantiate more than one, you're also asking for trouble.
     """
-    def __init__(self, duration, max_mem_mb = None):
+    # Activity history timeframe
+    _RESTART_TIMEFRAME = 60
+
+    def __init__(self, duration, max_mem_mb=None, max_resets=None):
```

Comment:
Super nitpicky, but in terms of interface I'd prefer the `max_resets` parameter to be a value per second, otherwise we don't really know what the value means exactly w/o reading the code.